component/LinkableText/LinkableText.tsx

```diff
@@ -0,0 +1,25 @@
+import Link from "next/link";
+
+const LinkableText = ({text}: {text: string}) => {
+    const sanitizeNewLine = text?.split('\n')
+    const regexUrl = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/
+
+    const textComponent = sanitizeNewLine?.map((textline, index) => {
+        const textLineSplit = textline?.split(' ')
+        return <p key={textline}>
+            {textLineSplit?.map((word, wordIndex) => {
+                if(wordIndex < textLineSplit?.length - 1) word += ' '
+                const hrefUrl = (word?.includes('http') || word?.includes('https')) ? word : `https://${word}`
+                if(regexUrl?.test(word)) return <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The current implementation assumes that if a word contains 'http' or 'https', it is a URL. This might not always be the case, as a word could contain these strings but not be a valid URL. Also, prepending `https://` to non-URL words could lead to incorrect links. Consider using a more robust method for URL detection.
```diff
-                const hrefUrl = (word?.includes('http') || word?.includes('https')) ? word : `https://${word}`
-                if(regexUrl?.test(word)) return <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link>
+                if(regexUrl?.test(word)) {
+                    const hrefUrl = word.startsWith('http') ? word : `https://${word}`;
+                    return <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link>
+                }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->